spec/features/search_page_spec.rb

```diff
@@ -308,4 +308,30 @@ def assert_data_set_length_is(count)
   def assert_search_results_headings(*headings)
     expect(all('h2 a').map(&:text)).to contain_exactly(*headings)
   end
+
+  scenario 'search filters are scoped to search query results' do
+    foo_dataset = DatasetBuilder
+                    .new
+                    .with_title('Foo')
+                    .with_publisher('Org 1')
+                    .with_topic({ name: 'fun', title: 'Fun' })
+                    .with_datafiles([{ format: 'CSV' }])
+                    .build
+
+    bar_dataset = DatasetBuilder
+                    .new
+                    .with_title('Bar')
+                    .with_publisher('Org 2')
+                    .with_topic({ name: 'trivia', title: 'Trivia' })
+                    .with_datafiles([{ format: 'PDF' }])
+                    .build
+
+    index(foo_dataset, bar_dataset)
+
+    search_for('foo')
+
+    expect(page).to have_select('Publisher', options: ['', 'Org 1'])
```

Comment:
I think it's probably worthing also checking what ISN'T in the select, as the point is to make sure that the bar_dataset related objects aren't shown in the dropdown.

Reply:
That's what the `options` attribute of the matcher does – https://www.rubydoc.info/github/jnicklas/capybara/Capybara/Node/Matchers#has_select%3F-instance_method

Reply:
Ah okay, in that case LGTM